src/libraries/System.Net.Primitives/src/System/Net/IPNetwork.cs

```diff
@@ -0,0 +1,400 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Binary;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+#pragma warning disable SA1648 // TODO: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3595
+
+namespace System.Net
+{
+    /// <summary>
+    /// Represents an IP network in CIDR notation with a <see cref="IPAddress"/> and a prefix length.
+    /// </summary>
+    /// <remarks>
+    /// This type disallows arbitrary IP/prefixLength CIDR pairs. <see cref="BaseAddress"/> must be defined so that all bits after the network prefix are set to zero.
+    /// In other words, <see cref="BaseAddress"/> is always the first usable address of the network.
+    /// The constructor and the parsing methods will throw in case there are non-zero bits after the prefix.
+    /// </remarks>
```

Comment:
This limitation makes sure that the data in the struct represents the network unambiguously, simplifying operations equality checking.

Note that it's not present in other implementations like aspnetcore's [IPNetwork type](https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/HttpOverrides/src/IPNetwork.cs).

To me it doesn't look like this can be a problem, since most use cases I've seen were providing the prefix address in this manner, and it's probably the best practice to do so, however I wonder if there are any other thoughts?

/cc @Tratcher